Sources/AuthFoundation/Security/Keychain.swift

```diff
@@ -48,27 +61,50 @@ public struct Keychain {
         public var value: Data
         
         /// Saves the item to the keychain.
-        public func save() throws {
-            let cfDictionary = query as CFDictionary
+        /// - Parameter authenticationContext: Optional `LAContext` to use when saving the credential, for platforms that support it.
+        /// - Returns: A ``Keychain/Item`` representing the new saved item.
+        @discardableResult
+        public func save(authenticationContext: KeychainAuthenticationContext? = nil) throws -> Item {
+            var cfDictionary = query
+            cfDictionary[kSecReturnAttributes as String] = kCFBooleanTrue
+            cfDictionary[kSecReturnData as String] = kCFBooleanTrue
+            cfDictionary[kSecReturnRef as String] = kCFBooleanTrue
+            
+            if let authenticationContext = authenticationContext {
+                cfDictionary[kSecUseAuthenticationContext as String] = authenticationContext
+            }
+            
+            if let accessControl = accessControl {
+                cfDictionary[kSecAttrAccessControl as String] = accessControl
+                cfDictionary.removeValue(forKey: kSecAttrAccessible as String)
+            }
 
-            Keychain.implementation.deleteItem(cfDictionary)
+            Keychain.implementation.deleteItem(cfDictionary as CFDictionary)
```

Comment:
Maybe just create explicit error for duplicate item(errSecDuplicateItem) case instead of physically delete? There is a risk that use may loose the data